test/unit/services/awsAdapter_spec.js

```diff
@@ -148,4 +152,146 @@ describe("The AwsAdapter class ", function () {
 			});
 		});
 	});
+
+	describe("terminating an instance", function () {
+
+		var describeInstancesStub;
+		var terminateInstancesStub;
+		var waitForStub;
+
+		describe("when the instance is running", function () {
+			var result;
+			var awsAdapter = new AwsAdapter(ec2);
+
+			before(function () {
+				describeInstancesStub = Sinon.stub(ec2, "describeInstancesAsync", function () {
+					var data = { Reservations : [ { Instances : [ { InstanceId : VALID_AWS_ID } ] } ] };
+					return Bluebird.resolve(data);
+				});
+
+				terminateInstancesStub = Sinon.stub(ec2, "terminateInstancesAsync", function () {
+					var data = { TerminatingInstances : [ { InstanceId : VALID_AWS_ID } ] };
+					return Bluebird.resolve(data);
+				});
+
+				waitForStub = Sinon.stub(ec2, "waitForAsync", function () {
+					var data = { Reservations : [ { Instances : [ {
+						InstanceId : VALID_AWS_ID,
+						state      : "terminated"
+					} ] } ] };
+					return Bluebird.resolve(data);
+				});
+
+				return awsAdapter.terminateInstances(VALID_ID)
+				.then(function (response) {
+					result = response;
```

Comment:
I don't think that this ever gets used in the assertions below. Do we care about this?